arrow-integration-test/src/schema.rs

```diff
@@ -178,10 +181,10 @@ mod tests {
                 ),
                 Field::new(
                     "c25",
-                    DataType::Struct(vec![
+                    DataType::Struct(Fields::from(vec![
```

Comment:
For anyone who uses `DataType::Struct` this is now getting complicated to construct

I wonder if we can ease the pain by having something 

```rust
impl DataType {
  fn new_struct(fields: impl Into<Fields>) -> Self {
 ..
}
```

So then this could be


```suggestion
                    DataType::new_struct(vec![
```